pkg/collector/corechecks/windows_event_log/check.go

```diff
@@ -0,0 +1,284 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build windows
+
+// Package evtlog defines a check that reads the Windows Event Log and submits Events
+package evtlog
+
+import (
+	"fmt"
+	"regexp"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	agentCheck "github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	agentEvent "github.com/DataDog/datadog-agent/pkg/metrics/event"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/api"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/api/windows"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/session"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/subscription"
+
+	"golang.org/x/sys/windows"
+)
+
+const checkName = "win32_event_log"
+
+// The lower cased version of the `API SOURCE ATTRIBUTE` column from the table located here:
+// https://docs.datadoghq.com/integrations/faq/list-of-api-source-attribute-value/
+const sourceTypeName = "event viewer"
+
+// Check defines a check that reads the Windows Event Log and submits Events
+type Check struct {
+	// check
+	core.CheckBase
+	config *Config
+
+	fetchEventsLoopWaiter sync.WaitGroup
+	fetchEventsLoopStop   chan struct{}
+
+	includedMessages []*regexp.Regexp
+	excludedMessages []*regexp.Regexp
+
+	// event metrics
+	eventPriority agentEvent.EventPriority
+
+	// event log
+	session             evtsession.Session
+	sub                 evtsubscribe.PullSubscription
+	evtapi              evtapi.API
+	systemRenderContext evtapi.EventRenderContextHandle
+	userRenderContext   evtapi.EventRenderContextHandle
+	bookmarkSaver       *bookmarkSaver
+}
+
+// Run executes the check
+func (c *Check) Run() error {
+	sender, err := c.GetSender()
+	if err != nil {
+		return err
+	}
+	// Necessary for check stats to be calculated (number of events collected, etc)
+	defer sender.Commit()
```

Comment:
comment says why. I can add a bit more context though.
```
	// Necessary for check stats to be calculated (number of events collected, etc)
	// Since events are collected in the background, this will update stats with the
	// count of events collected since the last Run() call.
```

Reply:
makes sense